From 64cd95df4937ea2b82023a6836851edc6c8d79c1 Mon Sep 17 00:00:00 2001
From: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
Date: Mon, 12 Nov 2018 15:24:27 +0100
Subject: [PATCH] PMCPESYS-104: regulator: tps65273: Don't apply constraints
 initially

When regulator is registered don't apply min/max voltage constraints.
Initial vout selector will be set a while later and we avoid crash on
some typ0 speed grade devices.

Signed-off-by: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
---
 drivers/regulator/tps65273-regulator.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/regulator/tps65273-regulator.c b/drivers/regulator/tps65273-regulator.c
index 8e7b7abe8be6..c787bfee22dd 100755
--- a/drivers/regulator/tps65273-regulator.c
+++ b/drivers/regulator/tps65273-regulator.c
@@ -64,7 +64,7 @@
 
 #define tps65273_reg_desc(num) {			\
 	.name		= "BUCK"#num,			\
-	.of_match	= of_match_ptr("BUCK"#num),	\
+	.of_match	= NULL,				\
 	.regulators_node = of_match_ptr("regulators"),	\
 	.id		= TPS65273_BUCK##num##_ID,		\
 	.ops		= &tps65273_ops,		\
@@ -204,6 +204,7 @@ static struct tps65273_platform_data *tps65273_parse_dt(struct device *dev)
 		}
 
 		init_data->constraints.valid_modes_mask = REGULATOR_MODE_NORMAL;
+		init_data->constraints.apply_uV = 0;
 	}
 
 	return pd;
